server/routes/locations.js

```diff
@@ -0,0 +1,44 @@
+const { isNilOrEmpty } = require('../utils/utils')
+const { processInput } = require('../services/validation')
+const { full } = require('../config/incident')
+
+module.exports = ({ locationService }) => {
+  // get list of all prisons
+  const viewPrisons = editMode => async (req, res) => {
+    const { token } = res.locals.user
+    const prisons = await locationService.getActiveAgenciesByType(token, 'INST')
+    const errors = req.flash('errors')
+    const data = { prisons, editMode, errors }
+    res.render('formPages/incident/changePrison', data)
+  }
+
+  // submit new prison id
+  const submit = (formName, editMode) => async (req, res) => {
+    const { bookingId } = req.params
+    const { agencyId } = req.body
+    const { username } = res.locals.user
+    const saveAndContinue = req.body.submit === 'save-and-continue'
+
+    if (saveAndContinue) {
```

Comment:
so you can get rid of this `processInput` thing and just have:
``` 
  if (!agencyId) {
        req.flash('errors', ...)
        return res.redirect(req.originalUrl)
    }
```

Reply:
done